### PR TITLE
Publish `CommunityToolkit.Maui.Core` NuGet Package

### PR DESCRIPTION
**Pull request by brminnick**
## Description

This Pull Request enables the publishing of our new NuGet Package, [`CommunityToolkit.Maui.Core`](https://github.com/CommunityToolkit/Maui/issues/213).

## Detailed Solution

- `CommunityToolkit.Maui.Core.csproj`
  - Remove `<IsPackable>false</IsPackable>`
- `azure-pipelines.yml`
  - `build_windows`
    - Add `Build and Pack CommunityToolkit.Maui.Core` Step
  - `build_macOS`
    - Add `Pack CommunityToolkit.Maui.Core NuGet` Step

### CODE REVIEW

**Comment by brminnick** [comment]
1st Draft submitted without modifying `azure-pipelines.yml`

**Comment by VladislavAntonyuk** [comment]
I think we need to change azure-pipeline.yml to build/test a solution rather than each project separately.

**Comment by VladislavAntonyuk** [comment]
And add ```<GeneratePackageOnBuild>True</GeneratePackageOnBuild>```

File: azure-pipelines.yml [hunk 1]
```diff
@@ -137,6 +143,10 @@ jobs:
...
-        displayName: 'Pack CommunityToolkit NuGets'
+        displayName: 'Pack CommunityToolkit.Maui.Core NuGet'
+        inputs:
+          script: 'dotnet pack $(PathToCommunityToolkitCoreCsproj) -c Release -p:PackageVersion=$(NugetPackageVersion) --output $(Build.ArtifactStagingDirectory)/nuget'
+      - task: CmdLine@2
+        displayName: 'Pack CommunityToolkit.Maui NuGet'
         inputs:
-          script: 'dotnet pack $(PathToCommunityToolkitCsproj) -c Release -p:PackageVersion=$(NugetPackageVersion) --output $(Build.ArtifactStagingDirectory)/nuget'
\ No newline at end of file
+          script: 'dotnet pack $(PathToCommunityToolkitCoreCsproj) -c Release -p:PackageVersion=$(NugetPackageVersion) --output $(Build.ArtifactStagingDirectory)/nuget'
```
Comment (VladislavAntonyuk) [inline azure-pipelines.yml:152]: looks like you pack Core multiple times. it should remain PathToCommunityToolkitCsproj

**Review by VladislavAntonyuk — DISMISSED** [review]

File: azure-pipelines.yml (hunk 1 above)
Comment (brminnick) [reply azure-pipelines.yml:152]: Whoops - good catch!

Comment (brminnick) [reply azure-pipelines.yml]: Fixed ✅

**Review by VladislavAntonyuk — APPROVED** [review]

**Pull request merged**